### DISCUSSION TITLE
Memoization (caching actions in one request)

### DISCUSSION THREAD

**Discussion by misog** [lorisleiva/laravel-actions#251] [Ideas]
Hi, it would be nice to have `::runMemoized()` for high performance. Also `::make()->handleMemoized()` or `::makeMemoized()->handleMemoized()` or `::makeMemoized()->handle()` could be implemented by various tricks.

About memoization:
https://github.com/spatie/once
https://github.com/laracraft-tech/memoize
https://github.com/solodkiy/memorize

Existing memoization libraries use object instance / stacktrace as part of cache key, so this is problematic for `::make()` which always creates new object. Maybe `static::class` with handle() parameters would be enough to implement cache keys.

**Comment by henzeb** [comment]
I' d rather see a concern like `HasMemoization`. then we do not have to call make or handleMemoized. we could have a proxy method that disables it for that particular call, like `withoutMemoization`
